standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnUtils.java

```diff
@@ -82,7 +82,29 @@ public static ValidTxnList createValidTxnListForCleaner(GetOpenTxnsResponse txns
     bitSet.set(0, abortedTxns.length);
     //add ValidCleanerTxnList? - could be problematic for all the places that read it from
     // string as they'd have to know which object to instantiate
-    return new ValidReadTxnList(abortedTxns, bitSet, highWaterMark, Long.MAX_VALUE);
+    return new ValidReadTxnList(abortedTxns, bitSet, highWatermark, Long.MAX_VALUE);
+  }
+
+  public static ValidTxnList createValidTxnListForAbortedTxnCleaner(GetOpenTxnsResponse txns, long minOpenTxn) {
```

Comment:
how is that different from `createValidTxnListForCleaner `, everything in Open_txns list `< minOpenTxn - 1` would be aborted

Reply:
I have renamed `createValidTxnListForCleaner` to `createValidTxnListForCompactionCleaner`. This is different from `createValidTxnListForAbortedTxnCleaner`, mainly that we dont truncate the abortedBits which seems unnecessary. We are also not concerned if there are open txns from other tables present in this list (open txn on the same table will obviously be handled since highWatermark will be updated to min open for that table - 1). We just create an exception list based on the highWatermark and use it for the creating the validWriteIdList.

Reply:
min open txn would be still capped by highWatermark in ValidReadTxnList constructor, result should be still the same

Reply:
Modified `createValidTxnListForCleaner` to handle both aborts in one single method. Added necessary Javadoc. Done.